examples/heif_convert.cc

```diff
@@ -380,27 +380,23 @@ int main(int argc, char** argv)
   // TODO: check, whether reading from named pipes works at all.
 
   std::ifstream istr(input_filename.c_str(), std::ios_base::binary);
-  uint8_t magic[12];
-  istr.read((char*) magic, 12);
-
-  if (heif_check_jpeg_filetype(magic, 12)) {
-    fprintf(stderr, "Input file '%s' is a JPEG image\n", input_filename.c_str());
+  uint8_t length[4];
+  istr.read((char*) length, 4);
+  uint32_t box_size = (length[0] << 24) + (length[1] << 16) + (length[2] << 8) + (length[3]);
+  if ((box_size < 16) || (box_size > 512)) {
+    fprintf(stderr, "Input file does not appear to start with a valid box length\n");
     return 1;
```

Comment:
I deliberately checked for JPEG here because there were several "bug reports" that claimed that a heic file could not be decoded when it was in fact a JPEG file.

Reply:
OK. That was a problem when I didn't have the whole file. However I can put it back in as part of passing the full number of bytes.

Reply:
Adapted to do a partial check based on the first four bytes.